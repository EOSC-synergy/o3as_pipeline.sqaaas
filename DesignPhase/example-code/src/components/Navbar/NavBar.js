import * as React from 'react';
import Button from '../views/landingPage/Sidebar/SectionComponents/inputs/Button';
import Sidebar from '../../views/landingPage/Sidebar/Sidebar';

/**
 * A component for the navigation of the website.
 */
export default function NavBar() {
    
    return (<div>
        <Button text="Edit Plot" onClick={props.handleSidebarOpen}/>
    </div>);
}