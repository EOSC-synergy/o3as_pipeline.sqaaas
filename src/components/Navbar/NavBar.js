import * as React from 'react';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import IconButton from '@mui/material/IconButton';
import EditIcon from '@mui/icons-material/Edit';
import { styled, useTheme } from '@mui/material/styles';
import AppBar from '@mui/material/AppBar';
import { indigo } from '@mui/material/colors';
import Link from '@mui/material/Link';
import Box from '@mui/material/Box';
import Menu from '@mui/material/Menu';
import MenuIcon from '@mui/icons-material/Menu';
import Button from '@mui/material/Button';
import MenuItem from '@mui/material/MenuItem';
import Logo from './Logo/Logo';

const pages = [
    {
        label: "About O3AS",
        href: "https://o3as.data.kit.edu/"
    },
    {
        label: "API",
        href: "https://api.o3as.fedcloud.eu/api/v1/ui/"
    },
    {
        label: "Legal",
        href: "#"
    }
];

const displayCollapsed = { xs: 'flex', md: 'none' };
const displayExpanded = { xs: 'none', md: 'flex' }


/**
 * A component for the navigation of the website.
 */
function Navbar(props) {

    const [anchorElNav, setAnchorElNav] = React.useState(null);

    const handleOpenNavMenu = (event) => {
        setAnchorElNav(event.currentTarget);
    };
    
    const handleCloseNavMenu = () => {
        setAnchorElNav(null);
    };

    return (
        <AppBar position="fixed" sx={{bgcolor: indigo[500]}}>
            <Toolbar disableGutters>
                {/* EXPANDED */}
                {/* Logo for expanded Navbar */}
                <Logo display={displayExpanded} />
                 {/* Menu for expanded Navbar */}
                 <Box sx={{ flexGrow: 1, display: displayExpanded }}>
                    {pages.map((page, idx) => (
                    <Button
                        key={idx}
                        sx={{ my: 2, color: 'white', display: 'block' }}
                    >
                        <a href={page.href} style={{textDecoration: 'inherit', color: 'inherit'}}>{page.label}</a>
                    </Button>
                    ))}
                </Box>

                {/* COLLAPSED */}
                {/* Menu for collapsed Navbar */}
                <Box sx={{ flexGrow: 1, display: displayCollapsed }}>
                    <IconButton
                        size="large"
                        aria-label="Menu"
                        aria-controls="menu-appbar"
                        aria-haspopup="true"
                        onClick={handleOpenNavMenu}
                        color="inherit"
                    >
                        <MenuIcon />
                    </IconButton>
                    <Menu
                        id="menu-appbar"
                        anchorEl={anchorElNav}
                        anchorOrigin={{
                            vertical: 'bottom',
                            horizontal: 'left',
                        }}
                        keepMounted
                        transformOrigin={{
                            vertical: 'top',
                            horizontal: 'left',
                        }}
                        open={Boolean(anchorElNav)}
                        onClose={handleCloseNavMenu}
                        sx={{
                            display: { xs: 'block', md: 'none' },
                        }}
                    >
                    {pages.map((page, idx) => (
                        <MenuItem
                            key={idx}
                            sx={{ my: 2, display: 'block' }}
                        >
                            <a href={page.href} style={{textDecoration: 'inherit', color: 'inherit'}}>{page.label}</a>
                        </MenuItem>
                    ))}
                    </Menu>
                </Box>
                {/* Logo for collapsed Navbar */}
                <Logo display={displayCollapsed} />

        </Toolbar>
                    
    </AppBar>);
}

export default Navbar;