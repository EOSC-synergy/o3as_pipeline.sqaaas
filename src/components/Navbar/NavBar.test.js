import React from 'react';
import ReactDOM from 'react-dom';
import Navbar from './Navbar';
import { render, screen } from '@testing-library/react';
import { within } from '@testing-library/dom';
import '@testing-library/jest-dom/extend-expect';
import { pages } from './Navbar';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Navbar />, div);
});

// Snapshot test
it('renders correctly', () => {
    const { asFragment } = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
});

it('renders all menu buttons in the expanded menu', () => {
    render(<Navbar />);
    const menuExpanded = screen.getByTestId('navbar-menu-expanded');
    const menuOptionsExpanded = within(menuExpanded).getAllByTestId(/menu-option/);
    expect(menuOptionsExpanded).toHaveLength(pages.length);

    for (let i = 0; i < pages.length; i++) {
        let textContent = new RegExp(pages[i].label);
        expect(menuOptionsExpanded[i]).toHaveTextContent(textContent);
        let testId = new RegExp(`menu-option-${pages[i].label}`)
        expect(within(menuExpanded).getByTestId(testId)).not.toBeDisabled();
    }
});

it('renders all menu buttons in the collpased menu', () => {
    render(<Navbar />);
    const menuCollapsed = screen.getAllByTestId('navbar-menu-collapsed');       // Mui renders multiple menus
    const menuOptionsCollapsed = within(menuCollapsed[0]).getAllByTestId(/menu-option/);
    expect(menuOptionsCollapsed).toHaveLength(pages.length);

    for (let i = 0; i < pages.length; i++) {
        let textContent = new RegExp(pages[i].label);
        expect(menuOptionsCollapsed[i]).toHaveTextContent(textContent);
        let testId = new RegExp(`menu-option-${pages[i].label}`)
        expect(within(menuCollapsed[0]).getByTestId(testId)).not.toBeDisabled();
    }
});
