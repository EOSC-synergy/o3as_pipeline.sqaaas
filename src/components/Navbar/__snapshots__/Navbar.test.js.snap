// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-8dh4wa-MuiPaper-root-MuiAppBar-root"
    data-testid="Navbar"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root"
      data-testid="Toolbar_Navbar"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiTypography-noWrap MuiLink-root MuiLink-underlineAlways css-1a3gl6d-MuiTypography-root-MuiLink-root"
        data-testid="logo-container"
        href="#"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1e0cb12-MuiGrid-root"
        >
          <img
            alt="logo"
            data-testid="logo-image"
            src="https://o3as.data.kit.edu/img/logos/o3as-logo.svg"
            style="height: 5em; margin: 0.5em;"
          />
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
            data-testid="logo-text"
          >
            O3AS Webapp
          </h6>
        </div>
      </a>
      <div
        class="MuiBox-root css-1t6c9ts"
        data-testid="navbar-menu-expanded"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1q39md6-MuiButtonBase-root-MuiButton-root"
          data-testid="navbar-expanded-menu-option-About O3AS"
          tabindex="0"
          type="button"
        >
          <a
            href="https://o3as.data.kit.edu/"
            style="text-decoration: inherit;"
          >
            About O3AS
          </a>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1q39md6-MuiButtonBase-root-MuiButton-root"
          data-testid="navbar-expanded-menu-option-API"
          tabindex="0"
          type="button"
        >
          <a
            href="https://api.o3as.fedcloud.eu/api/v1/ui/"
            style="text-decoration: inherit;"
          >
            API
          </a>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1q39md6-MuiButtonBase-root-MuiButton-root"
          data-testid="navbar-expanded-menu-option-Legal"
          tabindex="0"
          type="button"
        >
          <a
            href="#"
            style="text-decoration: inherit;"
          >
            Legal
          </a>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-1lvtzne"
      >
        <button
          aria-controls="menu-appbar"
          aria-haspopup="true"
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiTypography-noWrap MuiLink-root MuiLink-underlineAlways css-1a4kmyz-MuiTypography-root-MuiLink-root"
        data-testid="logo-container"
        href="#"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1e0cb12-MuiGrid-root"
        >
          <img
            alt="logo"
            data-testid="logo-image"
            src="https://o3as.data.kit.edu/img/logos/o3as-logo.svg"
            style="height: 5em; margin: 0.5em;"
          />
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
            data-testid="logo-text"
          >
            O3AS Webapp
          </h6>
        </div>
      </a>
    </div>
  </header>
</DocumentFragment>
`;
